### PR TITLE
BaSP-MR-153: Admin with firebase

### PR DESCRIPTION
**Pull request by francoax**
Development to respond [Trello-MR#153](https://trello.com/c/NBgtGxiQ)

### New
- Now the POST method add the new admin user in the Firebase db
- The DELETE method also delete the admin user from Firebase
- Implemented private route on admins routes

### Removed
- Logical delete from admins

### Fixes
- Updated the firebase uid in controllers and models of SuperAdmin, Admin, Trainer and member due to consistency
- Header fixed on middleware

### CODE REVIEW

File: src/controllers/admins.js [hunk 1]
```diff
@@ -1,59 +1,112 @@
...
-    })
-    .catch((error) => {
-      res.status(400).json({
-        message: 'An error ocurred!',
-        error: error.message,
-      });
+  let fireBaseUid;
+
+  try {
+    const adminExists = await Admin.findOne({ $or: [{ dni }, { email }] });
```
Comment (santimenendez98) [inline src/controllers/admins.js:14]: Maybe this part is not nessesary because firebase already verifies it

File: src/routes/index.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
 const express = require('express');
+const { default: verifyToken } = require('../middlewares/authMiddleware');
```
Comment (santimenendez98) [inline src/routes/index.js:2]: We agreed that routes will be implement in a separate ticket

File: src/routes/index.js [hunk 3]
```diff
@@ -11,7 +13,7 @@ const router = express.Router();
 
 router.use('/subscription', subscriptions);
 router.use('/trainer', trainers);
-router.use('/admins', admins);
+router.use('/admins', verifyToken, admins);
```
Comment (santimenendez98) [inline src/routes/index.js:16]: This part is also included in another ticket

**Review by santimenendez98 — CHANGES_REQUESTED** [review]

**Review by santimenendez98 — APPROVED** [review]
Good job!

**Review by FontanR — APPROVED** [review]
LGTM

**Pull request merged**